File: frontend/src/api-v2/users/contracts.ts
import { Role } from '@/models/shared/Role';

export type UserResponse = {
  id: string;
  email: string;
  firstName: string;
  lastName: string;
  roles: Role[];
  emailConfirmed: boolean;
  accepted: boolean;
};

export type UserWriteRequest = {
  email: string;
  firstName: string;
  lastName: string;
  role: string;
};

export type CruiseManagerOptionResponse = {
  id: string;
  email: string;
  firstName: string;
  lastName: string;
};

File: backend/ResearchCruiseApp/Api/Users/Lists/Contracts.cs
using ResearchCruiseApp.Infrastructure.Identity.Contracts;

namespace ResearchCruiseApp.Api.Users;

public sealed record UserResponse(
    Guid Id,
    string Email,
    string FirstName,
    string LastName,
    IList<string> Roles,
    bool EmailConfirmed,
    bool Accepted
)
{
    public static UserResponse From(UserDto user)
    {
        return new UserResponse(
            user.Id,
            user.Email,
            user.FirstName,
            user.LastName,
            user.Roles,
            user.EmailConfirmed,
            user.Accepted
        );
    }
}

public sealed record CruiseManagerResponse(Guid Id, string Email, string FirstName, string LastName)
{
    public static CruiseManagerResponse From(CruiseManagerOptionDto user)
    {
        return new CruiseManagerResponse(user.Id, user.Email, user.FirstName, user.LastName);
    }
}
